### DISCUSSION TITLE
Drawing multiple types of text (bold, colored, etc) in a single call?

### DISCUSSION THREAD

**Discussion by QuantumToasted** [dlemstra/Magick.NET#1723] [Help] [answered]
Hi there, I'm wondering if it is currently possible to draw something similar to the text in the below image via Magick.NET. I'm not even sure if it's possible in normal ImageMagick, so this may be a dead end, but I'm unsure what keywords to use to search for it, so I thought to start here.
![image](https://github.com/user-attachments/assets/1f8219c3-6cd2-4faf-b4b9-7481057cbc05)

I'm effectively trying to draw individual words in the text as bold and/or with a different color, while preserving the justification/alignment/etc of the text. I know this could theoretically be done with multiple text draw calls but it would be a nightmare to make it "look" like a normal sentence WRT alignment and justification.

**Comment by dlemstra** [comment, marked as answer]
You could use pango for this. You can read the text as an image and then composite this on top of another image.

```C#
        var readSettings = new MagickReadSettings
        {
            Font = "Courier New Ultra-Bold",
            FontPointsize = 42,
            BackgroundColor = MagickColors.None,
        };

        using var image = new MagickImage("pango:<span foreground='#ffac1c'>Trix</span><span foreground='#fff'> destroys</span><span foreground='#ffac1c'> Nimbus</span><span foreground='#fff'>'s supplies while he is asleep.</span>", readSettings);
```

You can find some docs about the syntax here: https://docs.gtk.org/Pango/pango_markup.html
